cmd/tests/tracing_module_test.go

```diff
@@ -236,6 +237,59 @@ func TestTracingInstrumentHTTP_FillsParams(t *testing.T) {
 	assertHasTraceIDMetadata(t, jsonResults)
 }
 
+func TestTracingInstrummentHTTP_SupportsMultipleTestScripts(t *testing.T) {
+	t.Parallel()
+
+	var gotRequests int64
+
+	tb := httpmultibin.NewHTTPMultiBin(t)
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
```

Comment:
nit: Having asserts like this is in general an anti-pattern and the testing package will give you warnings if you ever actually fail here. 

Although maybe with assert instead of require it is fine as it won't call [FailNow](https://pkg.go.dev/testing#T.FailNow) which is the usual problem :thinking:

Reply:
It works as I expected, though. I've just forced the test to fail at this statement, and I see no such warnings 🤔 

```
/home/oleiade/Dev/Grafana/k6/cmd/tests/tracing_module_test.go:250:
        	Error Trace:	/home/oleiade/Dev/Grafana/k6/cmd/tests/tracing_module_test.go:250
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2109
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2487
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2947
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:1991
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/asm_amd64.s:1594
        	Error:      	"00-1234567890-bf358b23435cbdd7-01" should have 55 item(s), but has 33
        	Test:       	TestTracingInstrummentHTTP_SupportsMultipleTestScripts
```

Reply:
Do you get a warning if you use `require` :thinking:

Reply:
Nope, same output

```
/home/oleiade/Dev/Grafana/k6/cmd/tests/tracing_module_test.go:250:
        	Error Trace:	/home/oleiade/Dev/Grafana/k6/cmd/tests/tracing_module_test.go:250
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2109
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2487
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:2947
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/server.go:1991
        	            				/home/oleiade/Dev/Grafana/k6/cmd/tests/asm_amd64.s:1594
        	Error:      	"00-1234567890-3d5a73e6627d9b8e-01" should have 55 item(s), but has 33
        	Test:       	TestTracingInstrummentHTTP_SupportsMultipleTestScripts
```

Reply:
Maybe I am mistaken and it only matters if it happens after the [test functions has returned](https://go.dev/play/p/cAtYXKMIOwBc) :thinking:  or if you have sub test and use the parent one :shrug: 

Whichever - it doesn't really matter. Sorry for the noise :bow: